system/modules/core/forms/FormFileUpload.php

```diff
@@ -108,8 +108,27 @@ public function validate()
 		$file = $_FILES[$this->strName];
 		$maxlength_kb = $this->getReadableSize($this->maxlength);
 
-		// Romanize the filename
-		$file['name'] = utf8_romanize($file['name']);
+		// Sanitize the filename
+		try
+		{
+			$file['name'] = \String::sanitizeFileName($file['name']);
+		}
+		catch (\InvalidArgumentException $e)
+		{
+			$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['filename'], $file['name']));
```

Comment:
As I already said on mumble before, do NOT output the filename as it is broken and the resulting message is unpredictable.